src/components/trends/cbg/RangeSelect.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * == BSD2 LICENSE ==
+ * Copyright (c) 2016, Tidepool Project
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the associated License, which is identical to the BSD 2-Clause
+ * License as published by the Open Source Initiative at opensource.org.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the License for more details.
+ *
+ * You should have received a copy of the License along with this program; if
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+import _ from 'lodash';
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import * as actions from '../../../redux/actions/';
+import LabeledCheckbox from '../../common/controls/LabeledCheckbox';
+
+import styles from './RangeSelect.css';
+
+export const RangeSelect = (props) =>
+  (<div className={styles.container}>
+    <LabeledCheckbox
+      checked={props.displayFlags.cbg100Enabled}
+      name="hundred"
+      label="100% of Readings"
+      onFn={_.partial(props.turnOnCbgRange, '100')}
+      offFn={_.partial(props.turnOffCbgRange, '100')}
+    />
+    <LabeledCheckbox
+      checked={props.displayFlags.cbg80Enabled}
+      name="eighty"
+      label="80% of Readings"
+      onFn={_.partial(props.turnOnCbgRange, '80')}
+      offFn={_.partial(props.turnOffCbgRange, '80')}
+    />
+    <LabeledCheckbox
+      checked={props.displayFlags.cbg50Enabled}
+      name="fifty"
+      label="50% of Readings"
+      onFn={_.partial(props.turnOnCbgRange, '50')}
+      offFn={_.partial(props.turnOffCbgRange, '50')}
+    />
+    <LabeledCheckbox
+      checked={props.displayFlags.cbgMedianEnabled}
+      name="median"
+      label="Median"
+      onFn={_.partial(props.turnOnCbgRange, 'median')}
+      offFn={_.partial(props.turnOffCbgRange, 'median')}
+    />
+  </div>);
+
+RangeSelect.propTypes = {
+  displayFlags: PropTypes.shape({
+    cbg100Enabled: PropTypes.bool.isRequired,
+    cbg80Enabled: PropTypes.bool.isRequired,
+    cbg50Enabled: PropTypes.bool.isRequired,
+    cbgMedianEnabled: PropTypes.bool.isRequired,
+  }).isRequired,
+  turnOnCbgRange: PropTypes.func.isRequired,
+  turnOffCbgRange: PropTypes.func.isRequired,
+  currentPatientInViewId: PropTypes.string.isRequired,
+};
+
+export function mapDispatchToProps(dispatch, ownProps) {
+  return bindActionCreators({
+    turnOnCbgRange: _.partial(
+      actions.turnOnCbgRange, ownProps.currentPatientInViewId
+    ),
+    turnOffCbgRange: _.partial(
+      actions.turnOffCbgRange, ownProps.currentPatientInViewId
+    ),
+  }, dispatch);
+}
+
+export default connect(
```

Comment:
A couple of comments here: the first and third functions can just be absent (can pass `null` for the first and just leave the third out - if you were looking at the `TrendsContainer` example, it adds an additional property in the third function (`{withRef: true}`) for compatibility with blip's parent-child component communication through refs (an outdated strategy), but nothing like that is necessary here). Generally `connect()`-ed components are also considered containers, not components.

Back in the baby days of redux, it wasn't considered very good practice to nest a lot of `connect()`-ed components, but apparently [the advice has changed a little](https://github.com/reactjs/react-redux/issues/159#issuecomment-148510079) (linking for Dan's comment, his link is now broken). However, it does sound like there are still [some potential performance issues](https://github.com/reactjs/redux/issues/125), for which there is [a potential middleware solution](https://github.com/acdlite/redux-batched-updates).

Since we're starting to be concerned about performance here in viz, I think it'd be good to discuss pros & cons of the options here and sync up on future strategy (with @hntrdglss as well). Also more generally the approach you've taken here is not what I expected. Since we already had code in blip that renders checkboxes for the BGM options and since the checkboxes have to be rendered in blip code due to the current split between what blip controls of the dataviz options, I expected that code to be reused (and there was a comment about this on the card ("...the checkboxes should use the same code as the checkboxes for the options available on BGM trends (Lines on/off, Range & Avg on/off, etc.)" from Nov. 8 when @darinkrauss was helping to get things more precise on the card). I get that the Trello cards are full of lots of info in a not easily digestible format, so I'm not surprised you missed that, but it might have been nice to talk through the approach before the PR was up to make sure we're all on the same page about approach and have any discussions about pros & cons of options with respect to future code.